moVirt/src/main/java/org/ovirt/mobile/movirt/rest/OVirtClient.java

```diff
@@ -703,35 +726,35 @@ private void updateClientBeforeCall() {
     public String login(String username, String password) {
         String token = "";
 
-        resetClientSettings();
-        setupVersionHeader("");
-        updateClientBeforeCall();
-        restClient.setRootUrl(authenticator.getBaseUrl());
-        try {
-            token = restClient.loginV4(username, password).getAccessToken();
-        } catch (Exception x) {// 405 Method Not Allowed - old API
+        synchronized (loginV4Lock) {
```

Comment:
Not sure if we really need synchronization here. One big downside is, if the user picks wrong ip address (in valid format), login is locked until timeout occurs (which can be for quite a long time).